client/components/mma/shared/asyncComponents/useAsyncLoader.ts

```diff
@@ -0,0 +1,37 @@
+import * as Sentry from '@sentry/browser';
+import { useEffect, useState } from 'react';
+import { trackEvent } from '../../../../utilities/analytics';
+import type ResponseProcessor from './ResponseProcessor';
+
+export default function useAsyncLoader<T>(
```

Comment:
As this is a hook it should probably live in `/utilities/hooks` with the others rather than in `/components`.